concent_api/concent_api/middleware.py

```diff
@@ -39,3 +49,81 @@ def __call__(self, request):
         response = self.get_response(request)
         response['Concent-Version'] = self._concent_version
         return response
+
+
+def should_use_html(request):
+    """
+    Function inspects request headers and determines is response should be in HTML format.
+    1) By default HTML should not be used (JSON is the default format)
+    2) If `text/html` is present in 'Accept' header and 'application/json' is not, HTML should be used
+    3) If both are present, HTML will be used if it has greater weight than JSON
+    """
+    def get_weight(selected_content_type, accepted_content_types):
+        for accepted_content_type in accepted_content_types.split(','):
+            if selected_content_type in accepted_content_type:
+                match = re.search(f"(?:{selected_content_type})(;q=(?P<weight>0.\d))?", accepted_content_type)
+                if match is not None and match.group('weight') is not None:
+                    return float(match.group('weight'))
+        return 1
+
+    def get_html_weight():
+        if 'headers' in request.META and 'Accept' in request.META['headers'] and \
+                any("text/html" in accepted for accepted in request.META['headers']['Accept'].split(',')):
+            return get_weight("text/html", request.META['headers']['Accept'])
+        return 0
```

Comment:
I think the `any("text/html" in accepted for accepted in request.META['headers']['Accept'].split(','))` part is superflous here. It may (insignificantly) speed things up when `text/html` is not present. It may slow things up when it is. I think this micro optimization is not really worth it.